src/plugins/newsfeed/public/components/newsfeed_header_nav_button.tsx

```diff
@@ -20,6 +20,11 @@
 import React, { useState, Fragment, useEffect } from 'react';
 import * as Rx from 'rxjs';
 import { EuiHeaderSectionItemButton, EuiIcon, EuiNotificationBadge } from '@elastic/eui';
+// eslint-disable-next-line
+import { PulseChannel } from 'src/core/server/pulse/channel';
```

Comment:
shouldn't it be the public dir instead of the server one?

Reply:
sadly types dont match up using public. i think public is missing `setup` and some other methods.

Reply:
That's why I think we should be explicit here. It's quite unlikely this component is initialised with the PulseChannel from the server. Importing it from public will ensure we don't accidentally call a method from server.